roman_numeral.js

```diff
@@ -0,0 +1,55 @@
+/* Convert the given number into a roman numeral.*/
+
+/**
+ * @param {int} num
+ * @return {array}
+ * depending on the spec, returns an array with num details
+ */
+function getNumDetails(num){
+  var n = num;
+  var arr = [];
+  var spec = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
+  for(var i = 0; i < spec.length; i++){
+    arr.push(parseInt(n / spec[i]));
+    n = n % spec[i];
+  }
+  return arr;
+}
+
+/**
+ * @param {string, int} romanLetter,numberOfTimes
+ * @return {string}
+ * returns a string containing `romanLetter` concatenated a `numberOfTimes`
+ */
+function concatRoman(romanLetter, numberOfTimes){
+  var result = '';
+  for(var i = 1; i <= numberOfTimes; i++){
+    result += romanLetter;
+  }
+  return result;
+}
+
+/**
+ * @param {array} numDetailed
+ * @return {string}
+ * returns a roman number based on `romanSpec` and `spec` from @getNumDetails
+ */
+function buildRoman(numDetailed){
+  var romanSpec = ['M','CM','D','CD','C','XC','L','XL','X','IX','V','IV','I'];
```

Comment:
So this needs to be synchronized with the `spec` defined in another scope. You should define both as part of the same data structure. Here a map of intValue -> romanSpec would be appropriate.
{1000: 'M', 900: 'CM'} but be careful that iterating over the keys will not be in order so you would need to sort them.
Also this is a constant and should be defined outside of the functions.